MM-27312 Allow uploading 10 files per post (#4214)
mattermost/docs: source/help/messaging/attaching-files.rst

Sharing Files
===============

With file attachments, you can share additional information that will help your team to visually understand your ideas. Sharing videos, voice recordings, screenshots, and photos will make your messages more effective and clear.

Attach Files to Messages
------------------------

You can attach files to messages in the following ways:

-  Use the attachment icon - select the paperclip icon inside the message input box

> -  Drag and Drop

Reviewer: ```suggestion
-  Drag and drop
```

-  Paste from the clipboard

Preview File Attachments
------------------------

Mattermost has a built-in file previewer that you can use to:

-  Download files
-  Share public links
-  View media

Select the thumbnail of an attached file to open it in the file previewer.

Downloading Files
~~~~~~~~~~~~~~~~~

You can download an attached file by selecting the download icon next to the file thumbnail.

Sharing Public Links
~~~~~~~~~~~~~~~~~~~~

Public URLs allow you to share attachments with anyone outside the Mattermost system. To share an attachment, select the thumbnail of an attachment, then select **Get Public Link**.

> .. tip::

Reviewer: I changed the formatting of this slightly; will need to confirm that it renders as expected in live.

  If **Get Public Link** is not visible in the file previewer, and you prefer that the feature is enabled, you can request your System Admin to enable the feature from the System Console under **Site Configuration > Public Links**.

> Viewing Media:

Reviewer: ```suggestion
Viewing Media
```

~~~~~~~~~~~~~~

The following media formats are supported on most browsers:

-  Images: BMP, GIF, JPG, JPEG, PNG, SVG
-  Video: MP4
-  Audio: MP3, M4A
-  Files: PDF, TXT

Other document previews (such as Word, Excel, or PPT) are not yet supported.

Attachment Limits and Sizes
---------------------------

> Mattermost Team Edition and Enterprise Edition supports a maximum of five (5) attached files per post. Mattermost Cloud supports a maximum of ten (10) attached files per post.

Reviewer: ```suggestion
Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post. Mattermost Cloud supports a maximum of 10 attached files per post.
```

Reviewer: I changed this based on the convention we use in our style guide as it's consistent with how we write numbers in the rest of our docs.

> Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post. Mattermost Cloud supports a maximum of 10 attached files per post.

Author: ```suggestion
Mattermost Cloud supports a maximum of 10 attached files per post. Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post, and this maximum will be increased to 10 files per post in a future release.
```

> Mattermost Cloud supports a maximum of 10 attached files per post. Mattermost Team Edition and Enterprise Edition support a maximum of five attached files per post, and this maximum will be increased to 10 files per post in a future release.

Reviewer 2: @cwarnermm @justinegeffen how are we handling the docs when it comes to cloud vs on-prem in general? I would find this statement strange as an on-prem admin. Perhaps docs should be versioned to allow admins to switch to the docs appropriate for their server version or cloud (default)

> this maximum will be increased to 10 files per post in a future release

Author: @esethna @justinegeffen @wiersgallak - The ability to version MM product documentation is a high priority project that will time to implement. This is the proposed interim approach that is intended to be temporary. This statement is scheduled to be removed as part of v5.31 documentation updates.

Author: @esethna The documentation now clarifies when the feature will be available for self-hosted deployments.
